File: CommonModels/Order.cs
using CommonModels.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace CommonModels
{
    public class Order
    {
        public Order()
        {
            Status = (int)OrderStatus.Ordered;
            TransactionId = Guid.NewGuid();
        }

        public int? Id { get; set; }
        public int? Status { get; set; }
        public Guid? TransactionId { get; set; }
        public int RestaurantId { get; set; }
        public int? UserId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public List<OrderProductsList> OrderProducts { get; set; } 
    }

    public class OrderProductsList
    {
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public int Price { get; set; }
    }
}

File: MyZomato-UI/src/app/core/models/order.ts
export interface IOrder{
    id:number,
    transactionId:string,
    status:string,
    //createdDate:Date,
    //modifiedDate:Date
    restaurantId:number,
    restaurant:string,
    amount:number,
    orderProducts:IOrderProduct[],
    isExpanded:boolean
}

export interface IOrderProduct{
    productName:string,
    productId:number,
    quantity:number
    price:number
}